plugins/instanceof-checkcast/src/main/java/org/qbicc/plugin/instanceofcheckcast/InstanceOfCheckCastBasicBlockBuilder.java

```diff
@@ -46,6 +51,103 @@ public InstanceOfCheckCastBasicBlockBuilder(final FactoryContext ctxt, final Bas
         this.ctxt = getContext();
     }
 
+    @Override
```

Comment:
These unreachable type optimizations make sense, but maybe we can need to come up with a more descriptive name for this BBB now.  (Then again, the same comment could be made about the entire plugin-in....).

Reply:
True. But it's possible these checks are not actually needed; I'll double-check.

Reply:
Yeah, they're not needed. This was part of an attempt to fix a reachability problem that I eventually resolved differently, so I've removed these checks.